app/config.py

```diff
@@ -9,6 +9,7 @@ class Config(object):
     SECRET_KEY = os.environ.get("SECRET_KEY")
 
     API_HOST_NAME = os.environ.get("API_HOST_NAME")
+    NOTIFY_RUNTIME_PLATFORM = os.environ.get("NOTIFY_RUNTIME_PLATFORM")
```

Comment:
Worth comparing this to the admin:

https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L99
https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L130

But then I'm confused about why we don't have anything similar in the API:
https://github.com/alphagov/notifications-api/blob/main/app/config.py

Nor anything in template preview:
https://github.com/alphagov/notifications-template-preview/blob/main/app/config.py

Any ideas?

Reply:
It looks like it was introduced in admin for the same reason, so it can run on ECS:
https://github.com/alphagov/notifications-admin//commit/77edc876cc63e474ce5c146be7dace50e1caf70c
However I'm not sure why the other apps don't need it.

Reply:
So it took me a bit of searching but I've got to the bottom of this.

**Setting the NOTIFY_RUNTIME_PLATFORM environment variable**
For all of our apps running in ECS we provide the environment variable `NOTIFY_RUNTIME_PLATFORM` in https://github.com/alphagov/notifications-aws/blob/main/terraform/modules/ecs-service/locals.tf#L41

**Consuming the NOTIFY_RUNTIME_PLATFORM environment variable**
The admin consumes that variable into its `app.config` in https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L99

Template preview doesn't consume that variable into its app.config, there is no corresponding line like admin

Antivirus doesn't consume that variable into its app.config, there is no corresponding line like admin

Document download frontend doesn't consume that variable into its app.config, but it will do with this PR you've put up

**Using the NOTIFY_RUNTIME_PLATFORM environment variable for whether to write logs to file**
Admin uses this environment variable because it has a version of utils higher than version 57.1.0 (https://github.com/alphagov/notifications-admin/blob/main/requirements.in#L31)

Template preview has a recent version of utils and so will try and determine whether to log to file based on app.config

Antivirus doesn't use this environment variable because it has a lower version of utils than 57.1.0 https://github.com/alphagov/notifications-antivirus/blob/main/requirements.in#L10

Document download does use this environment variable because it has a version of utils higher than version 57.1.0

**Setting the NOTIFY_LOG_PATH environment variable**
For all of our apps running in ECS we provide the environment variable `NOTIFY_LOG_PATH = application.log` in https://github.com/alphagov/notifications-aws/blob/main/terraform/modules/ecs-service/locals.tf#L46

**Consuming the NOTIFY_LOG_PATH environment variable**
The admin consumes that variable into its `app.config` in https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L27

Template preview consumes that variable into its `app.config` in https://github.com/alphagov/notifications-template-preview/blob/main/app/config.py#L54

Antivirus consumes that variable into its app.config in https://github.com/alphagov/notifications-antivirus/blob/main/app/config.py#L28

Document download frontend consumes the variable into its app.config

**What happens for each of the apps then**

Admin, it takes in the NOTIFY_LOG_PATH and the NOTIFY_RUNTIME_PLATFORM, passes them into a recent version of utils and decides not to do any logging to file

Antivirus, it takes in the NOTIFY_LOG_PATH of application.py and does attempt to write logs to file (because it has an old version of utils) and it appears to have permission to write those logs and is probably writing logs to file, even though we don't want it to

Template preview, it takes in the NOTIFY_LOG_PATH of application.py, and has a recent version of utils BUT it doesn't pull NOTIFY_RUNTIME_PLATFORM into its app.config file, so I'm not sure how template preview is working. I'd expect it to be erroring, saying I can't find app.config['NOTIFY_RUNTIME_PLATFORM]... 

## Conclusion
We have a lot of inconsistency across our apps. Not so nice. However I think this is the right direction for this pull request. I think you should make two small changes though which will mean the app config file will match the admins version as close as possible.

1. Set a fallback option of `paas` like in https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L99
2. Set the NOTIFY_RUNTIME_PLATFORM as `local` for `Development` like in https://github.com/alphagov/notifications-admin/blob/main/app/config.py#L130

cc @sakisv you might be interested in this stuff

Reply:
Nice investigation David. Indeed the reason that admin checks for the `NOTIFY_RUNTIME_PLATFORM` and the other apps don't is because I did the admin app first and had to do that to get it working. _I thought_ I put that code in utils, but apparently didn't. 

Might be worth putting that logic in there so that all the apps work consistently.

Reply:
Sorry Sakis, I don't quite understand what you've said there. Could you explain which code you thought you put in utils but apparently didn't please?

Reply:
Yeah, reading my comment again doesn't make it all that clear, sorry, let me clarify:

Right now in utils we're relying in `app.config["NOTIFY_RUNTIME_PLATFORM"]` being set ([link](https://github.com/alphagov/notifications-utils/blob/main/notifications_utils/logging.py#L78-L83)) and we configure the logs based on it.

What I'm suggesting and what I thought I had put in utils, is to check of whether the _environment variable_ is set, like we do in [admin](https://github.com/alphagov/notifications-admin/blob/2fcd8d01acebddb37f78173f46ea355cdfea468f/app/config.py#L99) and, not depend on each app having updated its `app.config`

That way we'll do all the checks in one place that's used from all the apps, instead of having to go and update each app.